### ISSUE TITLE
Roadmap for 1.0

### ISSUE THREAD

**Issue by tomchentw** [tomchentw/redux-component#3]
# Todos

* [ ] How about `this.refs` and `this.context`?
* [ ] `shouldComponentUpdate`
* [ ] Custom functions on Component.prototype

Please feel free to leave any thoughts in this thread.

**Comment by lapanoid** [comment]
@rk08 Hi! Awesome idea! I builded https://github.com/lapanoid/redux-cropper with this approach independently and wanted to extract some code in separate project, but found out your project in npm. Really want to help, now reading your code.  First thing that I really want is 'passing props api' for redux-component so it could be used as usual react component, the idea is that end user should not be aware that we use redux internally. What do you think?

**Comment by tomchentw** [comment]
@lapanoid I recently found the idea behind `redux-component` could be largely leveraged by [recompose](https://github.com/acdlite/recompose). Like to hear your feedback on this though.

What do you mean for `'passing props api'`? With `Componentize` of `redux-component`, the resulting component class  is unware of redux's existence so the user will not aware this.

**Comment by lapanoid** [comment]
@tomchentw I think my goal is different, in general I want redux app to be accessible as react component, and as far as I understand you 'reduxify' some pure components. It is like moving in opposite directions.

**Comment by tomchentw** [comment]
Got it. So this is what you're talking about:
https://github.com/lapanoid/redux-cropper/blob/master/src/index.js

That's correct. It's in the opposite direction. Just curious, have you read this?
https://github.com/erikras/ducks-modular-redux

**Comment by lapanoid** [comment]
@tomchentw Yes I saw that, this give some modularity within one application. I want something different, some components is too complex, so to reason about them we can you redux, so I basically want to use redux within redux much like [Dan's idea for redux-devtools](https://twitter.com/dan_abramov/status/612915178716401664), problem that in the end I will have app which is hard to compose, so I build this lib to convert redux app into react component https://github.com/lapanoid/redux-as-component.

**Comment by tomchentw** [comment]
Didn't quite get it. If you can use the generated `Cropper` React component without the knowledge of redux existence in [here](https://github.com/lapanoid/redux-cropper/blob/master/example/src/reactSimpleUsageApp.js#L75-L79), then I think the concept for `redux-as-component` is aligned with `redux-component`. But that differentiates what you mentioned above. Did I get something wrong?

**Comment by lapanoid** [comment]
From this point I guess so, we both have react component as the result. But as far as I understood you rely on pure functions and pure components all the way, but my goal is to convert arbitrary redux app to component without any limitations.
